### PR TITLE
ARROW-9532: [Python][Doc] Use Python3_EXECUTABLE instead of PYTHON_EXECUTABLE for finding Python executable

### PR DESCRIPTION
**Pull request by arw2019**

### CODE REVIEW

File: docs/source/developers/python.rst [hunk 1]
```diff
@@ -148,7 +148,7 @@ Pull in the test data and setup the environment variables:
    git submodule update
    export PARQUET_TEST_DATA="${PWD}/cpp/submodules/parquet-testing/data"
    export ARROW_TEST_DATA="${PWD}/testing/data"
-
+   popd
```
Comment (arw2019) [inline docs/source/developers/python.rst:151]: This is a nitpick: without it you need to do `popd` by hand before the next step

**Comment by kou** [comment]
Thanks.
Could you replace existing `PYTHON_EXECUTABLE` with `Python3_EXECUTABLE` and add note that "you need to use `PYTHON_EXECUTABLE` with CMake < 3.15" or something?
It's better that we assume that most users have newer CMake not older CMake.

**Comment by arw2019** [comment]
> Could you replace existing `PYTHON_EXECUTABLE` with `Python3_EXECUTABLE` and add note that "you need to use `PYTHON_EXECUTABLE` with CMake < 3.15" or something?
> It's better that we assume that most users have newer CMake not older CMake.

Done!

File: python/setup.py [hunk 2]
```diff
@@ -224,7 +224,7 @@ def _run_cmake(self):
             static_lib_option = ''
 
             cmake_options = [
-                '-DPYTHON_EXECUTABLE=%s' % sys.executable,
```
Comment (kou) [inline python/setup.py:227]: Ah, could you keep this?
If we can both `PYTHON_EXECUTABLE` and `Python3_EXECUTABLE`, it works on newer and older CMake.

**Review by kou — COMMENTED** [review]
Thanks.
Could you also replace `PYTHON_EXECUTABLE` at `docs/source/developers/python.rst` 308?

File: python/setup.py (hunk 2 above)
Comment (arw2019) [reply python/setup.py:227]: For sure!

**Review by kou — APPROVED** [review]
+1

Thanks!

**Comment by arw2019** [comment]
Thanks @kou for reviewing!

**Pull request closed without merging**